listenbrainz/webserver/views/player.py

```diff
@@ -55,3 +65,98 @@ def load():
+        mbid_clean = mbid.strip()
+        if not is_valid_uuid(mbid_clean):
+            raise BadRequest(f"Recording mbid {mbid} is not valid.")
+
+        recording_mbids.append(mbid_clean)
+
+    desc = request.args.get("desc", default="")
+    if not desc:
+        desc = "Instant playlist"
+
+    name = request.args.get("name", default="")
+    if not name:
+        name = "Instant playlist"
+
+    now = datetime.now()
+    playlist = WritablePlaylist(description=desc, name=name, creator="listenbrainz", creator_id=1, created=now)
+    for i, mbid in enumerate(recording_mbids):
+        rec = WritablePlaylistRecording(position=i, mbid=mbid, added_by_id=1, created=now)
+        playlist.recordings.append(rec)
+
+    fetch_playlist_recording_metadata(playlist)
+
+    return render_template(
+        "player/player-page.html",
+        props=ujson.dumps({"playlist": serialize_jspf(playlist)})
+    )
+
+
+@player_bp.route("/release/<release_mbid>/", methods=["GET"])
+def load_release(release_mbid):
+    """
+    This endpoint takes a release mbid, loads the tracks for this release and makes a playlist from it and
+    sends it to the front end via JSPF.
+
+    :statuscode 200: playlist generated
+    :statuscode 400: invalid recording_mbid arguments
+    """
+
+    release_mbid = release_mbid.strip()
+    if not is_valid_uuid(release_mbid):
+        raise BadRequest(f"Recording mbid {release_mbid} is not valid.")
+
+    playlist = None
+    if mb_engine:
+        release = get_release_by_mbid(release_mbid, includes=["media", "artists"])
+        if not release:
+            raise NotFound("This release was not found in our database. It may not have replicated to this server yet.")
+
+        name = "Release %s by %s" % (release["name"], release["artist-credit-phrase"])
+        desc = 'Release <a href="https://musicbrainz.org/release/%s">%s</a> by %s' % (release["mbid"],
+                                                                                      release["name"],
+                                                                                      release["artist-credit-phrase"])
+        now = datetime.now()
+        playlist = WritablePlaylist(description=desc, name=name, creator="listenbrainz", creator_id=1, created=now)
+        for medium in release["medium-list"]:
+            for recording in medium["track-list"]:
+                rec = WritablePlaylistRecording(title=recording["name"],
+                                                artist_credit=release["artist-credit-phrase"],
+                                                artist_mbids=[a["artist"]["mbid"] for a in recording["artist-credit"]],
+                                                release_name=release["name"],
+                                                release_mbid=release["mbid"],
+                                                position=recording["position"],
+                                                mbid=recording["mbid"],
```

Comment:
As far as I can tell this is a track MBID rather than a recording.
Load this page as an example, and click on a song title:
https://test.listenbrainz.org/player/release/da363688-9877-4cba-8ee0-89b42649e352

You'll get a 404. Changing the URL from `/recording/` to `/track/` gets a hit.

Reply:
Also if it's possible while you're in there, it would be great to add the track duration.

Reply:
Track durations are coming with the other metadata -- lets wait for that.